packages/amplify-e2e-tests/__tests__/api.test.ts

```diff
@@ -0,0 +1,41 @@
+require('../src/aws-matchers/'); // custom matcher for assertion
+import {
+  initProjectWithProfile,
+  deleteProject,
+  amplifyPush
+} from '../src/init';
+import { addApiWithSimpleModel } from '../src/categories/api';
+import { createNewProjectDir, deleteProjectDir, getProjectMeta } from '../src/utils';
+
+describe('amplify add api', () => {
+  let projRoot: string;
+  beforeEach(() => {
+    projRoot = createNewProjectDir();
+    jest.setTimeout(1000 * 60 * 60); // 1 hour
+  });
+
+  afterEach(async () => {
+    await deleteProject(projRoot);
+    deleteProjectDir(projRoot);
+  });
+
+  it('init a project and add the simple_model api', async () => {
+    await initProjectWithProfile(projRoot, { name: 'simplemodel' });
+    const meta = getProjectMeta(projRoot).providers.awscloudformation;
+    expect(meta.Region).toBeDefined();
+    const { AuthRoleName, UnauthRoleName, UnauthRoleArn, AuthRoleArn, DeploymentBucketName } = meta;
+
+    await expect(UnauthRoleName).toBeIAMRoleWithArn(UnauthRoleArn);
+    await expect(AuthRoleName).toBeIAMRoleWithArn(AuthRoleArn);
+    await expect(DeploymentBucketName).toBeAS3Bucket();
+
+    await addApiWithSimpleModel(projRoot, {});
+    await amplifyPush(projRoot);
+    const { output } = getProjectMeta(projRoot).api.simplemodel;
+    const { GraphQLAPIIdOutput, GraphQLAPIEndpointOutput, GraphQLAPIKeyOutput } = output;
+    
+    await expect(GraphQLAPIIdOutput).toBeDefined()
```

Comment:
Can we assert that these values we have are indeed the right ones using aws-sdk

Reply:
Added a todo in the code for additional checking but I think pulling from the CloudFormation outputs should be enough for now.